src/Thoth.Json/Decode.fs

```diff
@@ -805,17 +805,24 @@ module Decode =
     // This is used to force Fable use a generic comparer for map keys
     let private toMap<'key, 'value when 'key: comparison> (xs: ('key*'value) seq) = Map.ofSeq xs
 
-    let private autoObject (decoders: (string * BoxedDecoder)[]) (path : string) (value: obj) =
+    let private autoObject (decoders: (string * bool * BoxedDecoder)[]) (path : string) (value: obj) =
         if not (Helpers.isObject value) then
             (path, BadPrimitive ("an object", value)) |> Error
         else
-            (decoders, Ok []) ||> Array.foldBack (fun (name, decoder) acc ->
+            (decoders, Ok []) ||> Array.foldBack (fun (name, isOptional, decoder) acc ->
                 match acc with
                 | Error _ -> acc
                 | Ok result ->
-                    // TODO!!! Optional types shouldn't be required
-                    field name decoder path value
-                    |> Result.map (fun v -> v::result))
+                    if isOptional then
+                        // TODO: Copied from IOptionalGetter.Field. Should we unify it?
```

Comment:
If we can unify it using a function or an inline function I am for it.

We need to mark the function as internal or private so it is not exposed to users